xclim/sdba/adjustment.py

```diff
@@ -1204,6 +1246,270 @@ def _adjust(
+        ds = xr.Dataset(dict(ref=ref, hist=hist))
+        params = {
+            "quantiles": base_kws["nquantiles"],
+            "interp": adj_kws["interp"],
+            "extrapolation": adj_kws["extrapolation"],
+            "pts_dims": pts_dims,
+            "n_escore": n_escore,
+        }
+        out = mbcn_train(
+            ds, rot_matrices=rot_matrices, g_idxs=g_idxs, gw_idxs=gw_idxs, **params
+        )
+        params["group"] = base_kws["group"]
+
+        # postprocess
+        out["rot_matrices"] = rot_matrices
+        out["g_idxs"] = g_idxs
+        out["gw_idxs"] = gw_idxs
+
+        out.af_q.attrs.update(
+            standard_name="Adjustment factors",
+            long_name="Quantile mapping adjustment factors",
+        )
+        return out, params
+
+    def _adjust(
+        self,
+        sim: xr.DataArray,
+        ref: xr.DataArray,
+        hist: xr.DataArray,
+        *,
+        base: TrainAdjust = QuantileDeltaMapping,
+        base_kws_vars: dict[str, Any] | None = None,
+        adj_kws: dict[str, Any] | None = None,
+        period_dim=None,
+    ):
+        # set default values for non-specified parameters
+        base_kws_vars = base_kws_vars or {}
+        pts_dim = self.pts_dims[0]
+        for v in sim[pts_dim].values:
+            base_kws_vars.setdefault(v, {})
+            base_kws_vars[v].setdefault("group", self.group)
+            if isinstance(base_kws_vars[v]["group"], str):
+                base_kws_vars[v]["group"] = Grouper(base_kws_vars[v]["group"], 1)
+            if base_kws_vars[v]["group"] != self.group:
+                raise ValueError(
+                    f"`group` input in _train and _adjust must be the same."
+                    f"Got {self.group} and {base_kws_vars[v]['group']}"
+                )
+            base_kws_vars[v].pop("group")
+
+            base_kws_vars[v].setdefault("nquantiles", self.ds.af_q.quantiles.values)
+            if np.isscalar(base_kws_vars[v]["nquantiles"]):
+                base_kws_vars[v]["nquantiles"] = equally_spaced_nodes(
+                    base_kws_vars[v]["nquantiles"]
+                )
+            if "is_variables" in sim[pts_dim].attrs:
+                if "jitter_under_thresh_value" in base_kws_vars[v]:
+                    base_kws_vars[v]["jitter_under_thresh_value"] = str(
+                        convert_units_to(
+                            base_kws_vars[v]["jitter_under_thresh_value"],
+                            self.train_units[v],
+                            context="hydro",
+                        )
+                    )
+                if "adapt_freq_thresh" in base_kws_vars[v]:
+                    base_kws_vars[v]["adapt_freq_thresh"] = str(
+                        convert_units_to(
+                            base_kws_vars[v]["adapt_freq_thresh"],
+                            self.train_units[v],
+                            context="hydro",
+                        )
+                    )
```

Comment:
A consequence of using `multivar` is that the `thresh` are a bit more complicated to manage.
I could also unstack and restack variables inside the `mbcn_adjust`, I don't know which is preferable.